File: src/main/java/me/sullivan/glasslang/parser/nodes/ImportNode.java
package me.sullivan.glasslang.parser.nodes;

import me.sullivan.glasslang.interpreter.Interpreter;
import me.sullivan.glasslang.interpreter.primitives.types.Primitive;
import me.sullivan.glasslang.lexer.token.Token;

public class ImportNode extends Node
{
    public ImportNode(Token token)
    {
        super(token);
    }

    @Override
    public String getValue()
    {
        return token.getValue();
    }

    @Override
    public String toString()
    {
        return "ImportNode(" + getValue() + ")";
    }

    @Override
    public Primitive<?> visitor(Interpreter interpreter)
    {
        return interpreter.visitImportNode(this);
    }
}

File: src/me/sullivan/glasslang/parser/nodes/ImportNode.kt
package me.sullivan.glasslang.parser.nodes

import me.sullivan.glasslang.interpreter.Interpreter
import me.sullivan.glasslang.interpreter.primitives.Primitive
import me.sullivan.glasslang.lexer.token.Token

class ImportNode(token: Token) : Node(token)
{
    override val value: String
        get() = token.value

    override fun toString(): String
    {
        return "Import(${token.value})"
    }

    override fun visitor(interpreter: Interpreter): Primitive<*>?
    {
        return interpreter.visitImportNode(this)
    }
}